test/fixtures/es-module-loaders/example-loader.mjs

```diff
@@ -1,32 +1,52 @@
-import { URL } from 'url';
 import path from 'path';
-import process from 'process';
 import { builtinModules } from 'module';
 
 const JS_EXTENSIONS = new Set(['.js', '.mjs']);
 
 const baseURL = new URL('file://');
 baseURL.pathname = process.cwd() + '/';
 
-export function resolve(specifier, { parentURL = baseURL }, defaultResolve) {
+/**
+ * @param {string} specifier
+ * @param {{
+ *     parentURL:(string|undefined),
+ *     conditions:Array<string>,
+ * }} context
+ * @param {Function} defaultResolve
+ * @returns {{
+ *   url:string,
+ * }}
+ */
+export function resolve(specifier, context, defaultResolve) {
+  const { parentURL = baseURL } = context;
   if (builtinModules.includes(specifier)) {
     return {
       url: 'nodejs:' + specifier
     };
   }
-  if (/^\.{1,2}[/]/.test(specifier) !== true && !specifier.startsWith('file:')) {
+  if (!(specifier.startsWith('file:') || /^\.{1,2}[/]/.test(specifier))) {
     // For node_modules support:
     // return defaultResolve(specifier, {parentURL}, defaultResolve);
     throw new Error(
-      `imports must be URLs or begin with './', or '../'; '${specifier}' does not`);
+      "Imports must either be URLs or begin with './' or '../'; " +
```

Comment:
Note: in the current 13.x release, self-resolve specifiers are technically bare specifiers invalidating this check & statement.